core/sequencer/sequencer.go

```diff
@@ -77,18 +77,29 @@ func (s *Sequencer) TrackMasterships(ctx context.Context) {
 	s.tracker.Run(ctx)
 }
 
-// RunBatchForAllDirectories scans the directories table for new directories and creates new receivers for
-// directories that the sequencer is not currently receiving for.
-func (s *Sequencer) RunBatchForAllDirectories(ctx context.Context) error {
-	directories, err := s.directories.List(ctx, false)
+// AddAllDirectoriesForSequencing adds all directories to the set of resources
+// this sequencer attempts to obtain mastership for.
+func (s *Sequencer) AddAllDirectoriesForSequencing(ctx context.Context) error {
+	directories, err := s.directories.List(ctx, false /*deleted*/)
 	if err != nil {
 		return fmt.Errorf("admin.List(): %v", err)
 	}
 	for _, d := range directories {
-		knownDirectories.Set(1, d.DirectoryID)
-		s.tracker.AddResource(d.DirectoryID)
+		s.AddDirectoryForSequencing(d.DirectoryID)
 	}
+	return nil
+}
+
+// AddDirectoryForSequencing adds dirIDs to the set of resources this sequencer attempts to obtain mastership for.
+func (s *Sequencer) AddDirectoryForSequencing(dirIDs ...string) {
```

Comment:
Since the type is called `Sequencer`, maybe `ForSequencing` suffix is a bit redundant. How about renaming this to `AddDirectoryIDs`, and the above method to `AddAllDirectories`?